### ISSUE TITLE
[Metricbeat] Document Cluster Level vs. Node Level Metricsets

### ISSUE THREAD

**Issue by scottcrespo** [elastic/beats#7110]
## TL/DR
Whether the metricset is node-level or cluster-level is important.

In each metricset's documentation, the following information should be provided:

1. The endpoint called to retrieve the data with a link to the official documentation on that endpoint
2. Whether the metricset is node-level or cluster-level when the target is clustered
3. (Optional) Deployment notes/recommendations 

## Overview
This is a documentation request, with a specific documentation policy proposal. 

When monitoring a clustered service with metricbeat, it is not clear if a particular metricset is a node-level metricset or cluster-level metricset. How the metricset is obtained impacts how the beat must be deployed and the integrity of the data that's reported.

I realized this issue while deploying a variety of metricbeat modules to collect data on clustered services running on Kubernetes.

Currently, to determine if a metricset is node-level or cluster-level, I review the metricbeat source code to find the specific endpoint being called. Then I head to that technology's official docs to figure out if I must call each node individually, or if I can address the cluster as a whole. 

Determining if a metricset is node-level or cluster-level is important. For example, if the metricset is node-level but I treat it as cluster-level, I may just get a random node's report each time period, which results in inconsistent and incomplete data. Vice-versa, if I have a cluster-level metricset, but I treat it as node-level, I may receive the same cluster-wide, aggregated stats each time period *n* times. Then, I'm also likely to perform aggregations on the duplicate reports in Kibana, which results in a double aggregation, and potentially invalid results. 

You'll find that [metricbeat's official docs](https://www.elastic.co/guide/en/beats/metricbeat/current/running-on-kubernetes.html) touch on the need to deploy agents differently on Kubernetes (Daemonset vs. Deployment) depending on whether the metricset is node-level or cluster-level. 

Below I define node-level and cluster-level, and propose some specific documentation policies that might provide more information to users as to how metricbeat will collect data from a clustered service. 

### Node-level metricset

**Definition**
The metric beat should contact each node in the cluster **directly** to obtain data.

Therefore, deployment of a node-level metricset requires that a metricbeat agent be placed on every node in the cluster. 

**Deployment**
In the case of deploying on Kubernetes, you would likely create a *DaemonSet*.

**Example**
Mongo's `dbstats` metricset. Each node in a cluster may have different databases (in the case of sharded cluster), and may have different data volumes (as a result of replication and eventual consistency). 

### Cluster-Level metricset

**Definition**
A cluster-level metricset may contact one particular endpoint in the cluster (which may be an LB, master node, or any random node), and the response will either (a) report on each individual node, or (b) represent an aggregation of all nodes. 

**Deployment**
Deployment will likely single agent, which contacts a single endpoint. In the case of a Kubernetes deployment, this would be a *Deployment* with one replica.

**Example**
When retrieving data from the [RabbitMQ managmenet plugin](https://www.rabbitmq.com/management.html#http-api), all metricsets are cluster level.

## Documentation Proposal
In each metricset's documentation, the following information should be provided:

1. The endpoint called to retrieve the data with a link to the official documentation on that endpoint
2. Whether the metricset is node-level or cluster-level when the target is clustered
3. (Optional) Deployment notes/recommendations 

### Example

***

# MongoDB Module

### dbstats (beta)

**Description**
DbStats iterate through all databases on a particular node, and calls the db.stats() command.

**Endpoint**
[db.stats()](https://docs.mongodb.com/manual/reference/method/db.stats/)

**Node Level**
When operating a MongoDB cluster, it is recommend this metricset is treated as node-level

**Deployment Notes**
When collecting `dbstats` from a MongoDB cluster, metricbeat should be configured to connect to each node in the cluster directly.
